app/views/dashboard/_discovery_widget.html.erb

```diff
@@ -1,5 +1,5 @@
 <h4 class="ca"><%= _('Discovered Host Pool') %> - <%= @current_permission %></h4>
-<% all_hosts = Host::Discovered.search_for '' %>
+<% all_hosts = Host::Discovered.includes(:model).includes(:discovery_attribute_set).search_for '' %>
```

Comment:
I didn't see the `all_hosts.size` at the end, but we can still change the search_for maybe?